### PR TITLE
Adding support for LLaVA NeXT

### PR DESCRIPTION
**Pull request by yashaswikarnati**
# What does this PR do ?

added necessary task encoders for energon data module to support training LLaVA NeXT with NeVA Model. Also changes to neva base model to support passing num_image_tiles to mcore model.

**Collection**: [Note which collection this PR will affect]
Multimodal 

# Changelog 
- Task encoder for LLaVA NExT data module
- Modify data_step and forward_step of neva model to support image tiling.

# Usage
* Example usage of datamodule for LLaVa NeXT

```python
from nemo.collections.multimodal.data.energon.config import MultiModalSampleConfig
from nemo.collections.vlm import LlavaNextTaskEncoder
from nemo.collections.multimodal.data.energon import SimpleMultiModalDataModule
from transformers import AutoProcessor
processor = AutoProcessor.from_pretrained("llava-hf/llava-v1.6-mistral-7b-hf")
    data_path = args.data_path
    image_processor = processor.image_processor
    tokenizer = processor.tokenizer

    multimodal_sample_config = MultiModalSampleConfig()

    task_encoder = LlavaNextTaskEncoder(
        tokenizer=tokenizer, image_processor=image_processor, multimodal_sample_config=multimodal_sample_config
    )
    data = SimpleMultiModalDataModule(
        path=data_path,
        tokenizer=tokenizer,
        image_processor=image_processor,
        num_workers=8,
        micro_batch_size=mbs,
        global_batch_size=gbs,
        multimodal_sample_config=multimodal_sample_config,
        task_encoder=task_encoder,
    )
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/vlm/neva/model/base.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def neva_forward_step(model, batch) -> torch.Tensor:
         "attention_mask": batch.get("attention_mask", None),
         "loss_mask": batch.get("loss_mask", None),
         "labels": batch.get("labels", None),
+        "num_media_tiles": batch.get("num_media_tiles", 1),
```
Comment (yaoyu-33) [inline nemo/collections/vlm/neva/model/base.py:101]: Please default to None. I run into issues with 1. If none, it will create a correct tensor for it.

File: nemo/collections/multimodal/data/energon/base.py [hunk 2]
```diff
@@ -104,6 +104,14 @@ def __init__(
         self.train_dataloader_object = None
         self.val_dataloader_object = None
 
+    def io_init(self, **kwargs) -> fdl.Config[Self]:
```
Comment (yaoyu-33) [inline nemo/collections/multimodal/data/energon/base.py:107]: can you add docstring for this, explain why we have this?

File: nemo/collections/vlm/neva/data/llava_next_energon.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        Parameters:
+        input_sample (VQASample): The input VQA sample containing an image and conversation text.
+        output_sample (LlavaNextTextSample): The output sample structure where encoded results are stored.
+
+        Returns:
+        LlavaNextTextSample: The encoded output sample, containing processed tokens, labels,
+            images, loss masks, and metadata.
+        """
+        conversation_prompt = self.apply_prompt_template(input_sample)
+        logging.debug(f"task encoder encode_sample conversation_prompt {conversation_prompt}")
```
Comment (yaoyu-33) [inline nemo/collections/vlm/neva/data/llava_next_energon.py:84]: can you keep the logging consistent with nemo/collections/vlm/mllama/data/sample_encoder.py
         logging.debug(f"[Energon] task encoder encode_sample conversation_prompt {conversation_prompt}")
you put energon tag before
